tiledb/sm/query/readers/result_tile.h

```diff
@@ -1045,6 +1045,62 @@ class UnorderedWithDupsResultTile : public ResultTileWithBitmap<BitmapType> {
   }
 };
 
+/** Result tile used for ordered dimension labels. */
+class OrderedDimLabelResultTile : public ResultTile {
+ public:
+  /* ********************************* */
+  /*     CONSTRUCTORS & DESTRUCTORS    */
+  /* ********************************* */
+  OrderedDimLabelResultTile() = default;
+
+  OrderedDimLabelResultTile(
+      unsigned frag_idx,
+      uint64_t tile_idx,
+      const ArraySchema& array_schema,
+      const bool covered)
+      : ResultTile(frag_idx, tile_idx, array_schema)
+      , covered_(covered) {
+  }
+
+  /** Move constructor. */
+  OrderedDimLabelResultTile(OrderedDimLabelResultTile&& other) noexcept {
+    // Swap with the argument
+    swap(other);
+  }
+
+  /** Move-assign operator. */
+  OrderedDimLabelResultTile& operator=(OrderedDimLabelResultTile&& other) {
+    // Swap with the argument
+    swap(other);
+
+    return *this;
+  }
```

Comment:
Any reason we can't use the default move constructor and move assignment operators instead?

Reply:
https://app.shortcut.com/tiledb-inc/story/22087/result-tile-h-try-to-remove-the-custom-move-and-move-assign-constructors